simulacrum/dataset.py

```diff
@@ -1,185 +1,97 @@
-
-        :param length: The number of uuids.
-        :type length: int.
-        :return: The column of uuids.
-        :rtype: pd.Series
-
-        """
-        return pd.Series(list(map(lambda _: uuid4(), range(length))))
-
-    @staticmethod
-    def faker_data(ty, length):
-        """
-        Generate a column based on any faker data type.
-
-        :param ty: A configuration for the faker data. Must contain faker provider and related args as dict.
-        :param length: The number of rows wanted.
-        :param ty: dict.
-        :param length: The number of rows wanted.
-        :type length: int.
-        :return: The column of Faker data.
-        :rtype: pd.Series
-
-        """
-        try:
-            provider = ty["provider"]
-            del ty["provider"]
-            return pd.Series(list(map(lambda _: getattr(Faker(), provider)(**ty), range(length))))
-        except KeyError:
-            raise KeyError("You have to define the Faker provider.")
-        except AttributeError:
-            raise AttributeError("Faker().{}() is not a valid Faker provider.".format(provider))
-
-    def create(self, length, cols=None, types=None, coltypes=None):
-        series_res = {}
-        ops = {'num': self.num_data,
-               'int': self.num_int,
-               'norm': self.norm_data,
-               'exp': self.exp_data,
-               'bin': self.binom_data,
-               'pois': self.poisson_data,
-               'txt': self.text_data,
-               'name': self.name_data,
-               'addr': self.address_data,
-               'zip': self.zip_data,
-               'date': self.date_data,
-               'uuid': self.uuid_data,
-               'faker': self.faker_data}
-
-        if cols and types and coltypes:
-            logging.error('coltypes should not be defined when cols and types are defined')
-
-        if (cols and not types) or (types and not cols):
-            logging.error('cols and types must both be defined together, as lists')
-
-        if (cols and types):
-            validate_types(types)
-            if len(cols) != len(types):
-                logging.error('cols and types must be lists of equal length')
-            for i in len(cols):
-                series_res[col[i]] = ops[types[i]['type']](types[i], length)
-
-        else:
-            if not coltypes:
-                logging.error('please define either cols and types or coltypes')
-            # Assure iteritems compatibility throught 2.7 and 3+
-            try:
-                coltypes_items = coltypes.iteritems()
-            except AttributeError:
-                coltypes_items = coltypes.items()
-            for col, typ in coltypes_items:
-                data_builder = ops[typ['type']]
-                del typ['type']
-                series_res[col] = data_builder(typ, length)
+import logging
+import pandas as pd
 
-        return pd.DataFrame(series_res)
+from simulacrum import types
+
+TYPE_FUNCTIONS = {
```

Comment:
You make a convincing argument. If you want to make some changes and so long as they are testable, I will merge. We should also let @theianrobertson chime in. My other request is can we add some handlers so that legacy uses will still function? If not can we make sure it is clearly documented in the README.md Make sure we also update the Changelog and the setup.cfg so we can cut a new version once tested and merged.